# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/nsContentUtils.cpp
# Commit: 9abe42390b6c
# Full Hash: 9abe42390b6c1bb52dc2ba7acdd16a71f86b8b89
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2019-06-12 12:02:49
# Description:
#   Bug 1558412 - Make `nsContentUtils::GetHTMLEditor()` null-check of the given `nsPresContext*` r=smaug
#   
#   Surprisingly, `Document::ExecCommand()` may be called when there is no
#   corresponding `nsPresContext`.  Then, `nsContentUtils::GetHTMLEditor()`
#   crashes because it does not check whether the argument is `nullptr` or not.
# ==============================================================================

diff -r 9b7d0410ba13 -r 9abe42390b6c dom/base/nsContentUtils.cpp
--- a/dom/base/nsContentUtils.cpp	Tue Jun 11 14:30:51 2019 +0000
+++ b/dom/base/nsContentUtils.cpp	Wed Jun 12 01:58:01 2019 +0000
@@ -6778,6 +6778,10 @@
 }
 
 HTMLEditor* nsContentUtils::GetHTMLEditor(nsPresContext* aPresContext) {
+  if (!aPresContext) {
+    return nullptr;
+  }
+
   nsCOMPtr<nsIDocShell> docShell(aPresContext->GetDocShell());
   bool isEditable;
   if (!docShell || NS_FAILED(docShell->GetEditable(&isEditable)) || !isEditable)
